main/views.py

```diff
@@ -449,6 +452,27 @@ def delete(self, request, *args, **kwargs):
         return super(DeleteView, self).delete(request, *args, **kwargs)
 
 
+class UserMediaView(LoginRequiredMixin, generic.View):
+    """Respond with a given user uploaded file if the request
+    is logged in. Makes it so that user uploads are no longer
+    public."""
+
+    def get_response_file(self, filename):
+        filepath = os.path.join(settings.MEDIA_ROOT, filename)
```

Comment:
As noted in the chat, this is dangerous. An attacker can craft a filename that will result in arbitrary file access on the server. (outside MEDIA_ROOT)

It's not a OMG WE ARE DOOMED scenario, we're still behind a LoginRequiredMixin and Apache seems to be filtering out the more simple malicious filenames, but I still really don't want to rely on those protections alone. Especially as the fix is one if statement ;)

Thus, there should be a check for making sure that the resulting filepath is still inside MEDIA_ROOT.

Reply:
For those interested, it's called a path traversal attack:
https://owasp.org/www-community/attacks/Path_Traversal